src/app/clusters/thread-network-directory-server/DefaultThreadNetworkDirectoryStorage.cpp

```diff
@@ -46,7 +46,8 @@ CHIP_ERROR DefaultThreadNetworkDirectoryStorage::StoreIndex()
 {
     VerifyOrDie(mInitialized);
     StorageKeyName key = DefaultStorageKeyAllocator::ThreadNetworkDirectoryIndex();
-    return mStorage.SyncSetKeyValue(key.KeyName(), mExtendedPanIds, mCount * ExtendedPanId::size());
+    static_assert(kCapacity * ExtendedPanId::size() <= UINT16_MAX); // kCapacity >= mCount
```

Comment:
What's the `// kCapacity >= mCount` comment?

Reply:
It’s pointing out why the static assert makes sure the static cast in the next line won’t truncate